src/main/java/uk/gov/hmcts/darts/authentication/controller/impl/AuthenticationExternalUserController.java

```diff
@@ -1,22 +1,32 @@
 package uk.gov.hmcts.darts.authentication.controller.impl;
 
+import com.nimbusds.jwt.SignedJWT;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.servlet.ModelAndView;
 import uk.gov.hmcts.darts.authentication.controller.AuthenticationController;
+import uk.gov.hmcts.darts.authentication.exception.AuthenticationError;
+import uk.gov.hmcts.darts.authentication.model.SecurityToken;
 import uk.gov.hmcts.darts.authentication.service.AuthenticationService;
+import uk.gov.hmcts.darts.authorisation.service.AuthorisationService;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
 
 import java.net.URI;
+import java.text.ParseException;
+import java.util.List;
 
 @Slf4j
 @RestController
 @RequestMapping("/external-user")
 @RequiredArgsConstructor
 public class AuthenticationExternalUserController implements AuthenticationController {
 
+    private static final String EMAILS_CLAIM_NAME = "emails";
+
     private final AuthenticationService authenticationService;
+    private final AuthorisationService authorisationService;
```

Comment:
As `AuthorisationService` lives in a seaparte feature, we shouldn't access it directly. Instead, we should expose/wrap the needed functionality via an "api" interface within the `authorisation` feature. Any code in `authentication` would then use that api interface to call the needed functionality.

This pattern is intended to make inter-feature coupling explict and manageable. I will add a section to the Best Practices document on this.

Reply:
@chrisbellingham-hmcts Done in commit f466057